### PR TITLE
Exit with an error code if command parsing fails

### PR DESCRIPTION
**Pull request by tobim**
This PR also changes the option `--config-file` to `--config`.

### CODE REVIEW

File: libvast/src/error.cpp [hunk 1]
```diff
@@ -72,6 +73,9 @@ std::string render(caf::error err) {
   using caf::atom_uint;
   std::ostringstream oss;
   auto category = err.category();
+  if (atom_uint(category) == atom_uint("vast")
+      && static_cast<vast::ec>(err.code()) == ec::silent)
```
Comment (mavam) [inline libvast/src/error.cpp]:
```suggestion
      && err == ec::silent)
```

I haven't tested this, but I think this is possible.

File: libvast/src/error.cpp [hunk 2]
```diff
@@ -74,7 +74,7 @@ std::string render(caf::error err) {
   std::ostringstream oss;
   auto category = err.category();
   if (atom_uint(category) == atom_uint("vast")
-      && static_cast<vast::ec>(err.code()) == ec::silent)
+      && err == ec::silent)
```
Comment (mavam) [inline libvast/src/error.cpp]: And now it fits on the previous line as well.

**Comment by dominiklohmann** [comment]
> Adding a new error code seems like a one-off here. Can't we just check for `ec::unspecified` in combination with an empty message?

Wouldn't that still print the !! in the beginning? I like the explicitness of the new error code.

**Comment by tobim** [comment]
Yes, it is intentionally explicit.

**Comment by mavam** [comment]
I'm approving because I can see the argument on the other side. My point was that if we would add a new error code whenever we need some dispatching logic in a different context, we'd abuse the error abstraction. The error code should represent a *class* of errors, not a single one. However, `silent` is generic enough to serve as such a class.

**Pull request merged**